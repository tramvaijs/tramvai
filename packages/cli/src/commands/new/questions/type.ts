import any from '@tinkoff/utils/array/any';

type Type = 'app' | 'child-app';

const choices = [
  {
    name: 'Standalone tramvai app',
    value: 'app',
  },
  {
    name: 'Microfrontend implemented by tramvai child-app',
    value: 'child-app',
  },
];

const typeQuestion = (answer: string) => ({
  type: 'list' as const,
  name: 'type' as const,
  message: 'Choose a type for your project',
  choices,
  when: () => !any<{ name: string; value: string }>((choice) => choice.value === answer)(choices),
  default: 'app' as const,
});

export { Type, typeQuestion };
