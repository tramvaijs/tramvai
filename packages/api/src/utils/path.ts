import path from 'node:path';

export const resolveAbsolutePathForFile = ({
  file,
  sourceDir,
  rootDir,
}: {
  file: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(file)) {
    return file;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, file);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, file);
  }
  return path.resolve(rootDir, sourceDir, file);
};

export const resolveAbsolutePathForFolder = ({
  folder,
  rootDir,
}: {
  folder: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(folder)) {
    return folder;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, folder);
  }
  return path.resolve(rootDir, folder);
};
