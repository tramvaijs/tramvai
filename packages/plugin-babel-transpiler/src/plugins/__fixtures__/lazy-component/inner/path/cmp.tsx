import React from 'react';

const InnerCmp = () => {
  return <div>inner cmp</div>;
}

export default InnerCmp;
