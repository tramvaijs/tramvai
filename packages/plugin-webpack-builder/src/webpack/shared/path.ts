import path from 'node:path';

export const resolveAbsolutePathForEntry = ({
  entry,
  sourceDir,
  rootDir,
}: {
  entry: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(entry)) {
    return entry;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, entry);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, entry);
  }
  return path.resolve(rootDir, sourceDir, entry);
};

export const resolveAbsolutePathForOutput = ({
  output,
  rootDir,
}: {
  output: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(output)) {
    return output;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, output);
  }
  return path.resolve(rootDir, output);
};
