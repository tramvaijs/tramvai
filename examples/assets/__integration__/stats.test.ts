import path from 'path';
import type { StartCliResult } from '@tramvai/test-integration';
import { startCli } from '@tramvai/test-integration';

jest.setTimeout(30000);

describe('assets', () => {
  let app: StartCliResult;

  beforeAll(async () => {
    app = await startCli('assets', {
      rootDir: path.resolve(__dirname, '../'),
    });
  }, 80000);

  afterAll(() => {
    return app.close();
  });

  it('runtime chunk snapshot', async () => {
    const statsJSONResponse = await fetch(`${app.staticUrl}/dist/client/stats.json`);
    const statsJSON = await statsJSONResponse.json();

    delete statsJSON.outputPath;
    delete statsJSON.publicPath;

    statsJSON.chunks = statsJSON.chunks.map(
      (chunk: {
        hash: string;
        rendered?: boolean;
        sizes?: Record<string, number>;
        size?: number;
      }) => {
        chunk.hash = '0';

        // remove unstable fields
        delete chunk.rendered;
        delete chunk.sizes;
        delete chunk.size;

        return chunk;
      }
    );

    expect(JSON.stringify(statsJSON, null, 2)).toMatchSnapshot();
  });
});
