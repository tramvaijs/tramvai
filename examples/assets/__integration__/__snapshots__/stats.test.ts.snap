// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assets runtime chunk snapshot 1`] = `
"{
  "assetsByChunkName": {
    "platform": [
      "platform.js"
    ],
    "runtime": [
      "runtime.js"
    ],
    "@_routes_index": [
      "@_routes_index.chunk.js"
    ],
    "react": [
      "react.js"
    ]
  },
  "namedChunkGroups": {
    "platform": {
      "name": "platform",
      "chunks": [
        "runtime",
        "react",
        "platform"
      ],
      "assets": [
        {
          "name": "runtime.js"
        },
        {
          "name": "react.js"
        },
        {
          "name": "platform.js"
        }
      ],
      "filteredAssets": 0,
      "assetsSize": null,
      "filteredAuxiliaryAssets": 0,
      "auxiliaryAssetsSize": 0
    },
    "@_routes_index": {
      "name": "@_routes_index",
      "chunks": [
        "@_routes_index"
      ],
      "assets": [
        {
          "name": "@_routes_index.chunk.js"
        }
      ],
      "filteredAssets": 0,
      "assetsSize": null,
      "filteredAuxiliaryAssets": 2,
      "auxiliaryAssetsSize": null
    }
  },
  "entrypoints": {
    "platform": {
      "name": "platform",
      "chunks": [
        "runtime",
        "react",
        "platform"
      ],
      "assets": [
        {
          "name": "runtime.js"
        },
        {
          "name": "react.js"
        },
        {
          "name": "platform.js"
        }
      ],
      "filteredAssets": 0,
      "assetsSize": null,
      "filteredAuxiliaryAssets": 0,
      "auxiliaryAssetsSize": 0
    }
  },
  "chunks": [
    {
      "initial": false,
      "entry": false,
      "recorded": false,
      "names": [
        "@_routes_index"
      ],
      "idHints": [],
      "runtime": [
        "runtime"
      ],
      "files": [
        "@_routes_index.chunk.js"
      ],
      "auxiliaryFiles": [
        "bd514a4fdeae248aabd3844c43fa2cc1.svg",
        "bf9fb7baf3f9985e84328ccf5eeef509.svg"
      ],
      "hash": "0",
      "childrenByOrder": {},
      "id": "@_routes_index"
    },
    {
      "initial": true,
      "entry": false,
      "recorded": false,
      "names": [
        "platform"
      ],
      "idHints": [],
      "runtime": [
        "runtime"
      ],
      "files": [
        "platform.js"
      ],
      "auxiliaryFiles": [],
      "hash": "0",
      "childrenByOrder": {},
      "id": "platform"
    },
    {
      "initial": true,
      "entry": false,
      "recorded": false,
      "reason": "split chunk (cache group: reactCacheGroup) (name: react)",
      "names": [
        "react"
      ],
      "idHints": [
        "reactCacheGroup"
      ],
      "runtime": [
        "runtime"
      ],
      "files": [
        "react.js"
      ],
      "auxiliaryFiles": [],
      "hash": "0",
      "childrenByOrder": {},
      "id": "react"
    },
    {
      "initial": true,
      "entry": true,
      "recorded": false,
      "names": [
        "runtime"
      ],
      "idHints": [],
      "runtime": [
        "runtime"
      ],
      "files": [
        "runtime.js"
      ],
      "auxiliaryFiles": [],
      "hash": "0",
      "childrenByOrder": {},
      "id": "runtime"
    }
  ]
}"
`;
